### ISSUE TITLE
Allow entry point to not be HTML? 

### ISSUE THREAD

**Issue by molant** [sonarwhal/sonarwhal#550]
Original conversation [in here](https://github.com/sonarwhal/sonar/pull/547#discussion_r141770380).

Right now all `sonar` documentation talks about analyzing website, all the examples we have scan a url that points to an HTML, etc.

@alrra wants to be capable of point to something that is not an HTML page and get related results. For example, point to an image and if it is a png but the response says it's a jpeg tell the user.

I can see this as a useful scenario but not must have at the moment. Once we start working on the code editor plugin things could change because we should be capable of analyzing whatever the user is working on (HTML, JS, CSS, etc.).

The current status is that:

* there are rules that make assumptions of what is returned on `targetfetch::end` (an HTML) and other's that don't and do a check to know what it is. If we decide to not assume anything, then we should do that check only once instead of on each rule. Maybe at the connector level and add the information to the event.
* all our documentation is written in such a way that user might think we only analyze websites
* there isn't a single test on the connectors where the entry url isn't HTML

@sonarwhal/core thoughts? something else I'm missing?

**Comment by alrra** [comment]
> wants to be capable of point to something that is not an HTML page and get related results. 

Yes, basically allow `sonar` to accept any URL (or file type) and be smart enough to execute only what is needed for that particular type of resource.

> If we decide to not assume anything, then we should do that check only once instead of on each rule. Maybe at the connector level and add the information to the event.

:+1:

**Comment by alrra** [comment]
Closing as this issue is no longer relevant.

See also: https://github.com/sonarwhal/sonarwhal/commit/f4536396d3177c969eee023a6598aa1ba5e0cefe.
